index.ts

```diff
@@ -66,28 +97,33 @@ async function sendGroupEvent(invoice, customer, due_last_month, due_total, paid
     })
 }
 
-function last_month(invoices, key) {
+function last_month(global, invoices: StoredInvoice[], key) {
     const today = new Date()
     const firstDayThisMonth = Math.floor(new Date(today.getFullYear(), today.getMonth(), 1) / 1000)
     const firstDayNextMonth = Math.floor(new Date(today.getFullYear(), today.getMonth() + 1, 1) / 1000)
     return invoices
-        .filter(({ period_end }) => {
-            return period_end > firstDayThisMonth && period_end < firstDayNextMonth
+        .filter((invoice) => {
+            const timestamp = global.getInvoiceTimestamp(invoice)
+            return (
+                timestamp !== undefined  // older invoices may not have all timestamp data
```

Comment:
This could also result in this function returning `0` when there _is_ data, but we just don't have it. We could backfill it, but that would introduce more complexity here -- not sure if it's worth it?